### ISSUE TITLE
Loss too high and the accuracy remains constant even after 20 epochs

### ISSUE THREAD

**Issue by adimyth** [tflearn/tflearn#873]
I tried to train the LSTM model using the below code on the amazon_cells_labelled.txt. 
The dataset contains 1000 cases with the sentences and their corresponding labels seperated by a 'tab'.
However,when I run the model I get a constant loss value around 0.7 to 0.8 & the accuracy remains the same around 0.49 to 0.51.
I wonder what's if anything's wrong with the code or initialisation or my understanding about the LSTM.
Someone please guide me!!

`import os
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
import pandas as pd
import tflearn
import numpy as np
import random
import string
import collections
from tflearn.data_utils import pad_sequences,to_categorical
from sklearn.model_selection import train_test_split

X = []
y = []
with open('amazon_cells_labelled.txt') as f:
	for line in f:
		sentence,label = line.split('\t')
		X.append(sentence)
		y.append(label)

allWords = []
allDocuments = []
allLabels = []

def collect_words(X):
	for line in X:
		line = line.lower().encode('utf-8')
		words = line.split()
		for word in words:
			word = word.translate(None,string.punctuation)
			if word!='':
				allWords.append(word)

def convert_words_to_integers(dictionary,X,allDocuments,allLabels,y):
	document = []
	for i,line in enumerate(X):
		line = line.lower().encode('utf-8')
		words = line.split()
		for word in words:
			word = word.translate(None,string.punctuation)
			if word in dictionary:
				index = dictionary[word]
			else:
				index = 0
			document.append(index)
		allDocuments.append(document)
		allLabels.append(y[i])

vocabulary_size = 10000

def build_dataset(words):
	count = [['UNK',-1]]
	count.extend(collections.Counter(words).most_common(vocabulary_size - 1))
	dictionary = dict()
	for word,_ in count:
		dictionary[word] = len(dictionary)
	data = list()
	unk_count = 0
	for word in words:
		if word in dictionary:
			index = dictionary[word]
		else:
			index = 0
			unk_count = unk_count + 1
		data.append(index)
		count[0][1] = unk_count
		reverse_dictionary = dict(zip(dictionary.values(),dictionary.keys()))
		return dictionary,reverse_dictionary

collect_words(X)
print(len(allWords))

dictionary,reverse_dictionary = build_dataset(allWords)
del(allWords)
len(dictionary)

convert_words_to_integers(dictionary,X,allDocuments,allLabels,y)
print(len(allDocuments))
print(len(allLabels))

c = list(zip(allDocuments,allLabels))
random.shuffle(c)
allDocuments,allLabels = zip(*c)

train_X,test_X,train_y,test_y = train_test_split(allDocuments,allLabels,test_size = 0.3,random_state = 42)

train_X = pad_sequences(train_X,maxlen = 200,value = 0.)
test_X = pad_sequences(test_X,maxlen = 200,value = 0.)
train_y = to_categorical(train_y,nb_classes = 2)
test_y = to_categorical(test_y,nb_classes = 2) 

net = tflearn.input_data(shape = [None,200])
net = tflearn.embedding(net,input_dim = vocabulary_size,output_dim = 100)
net = tflearn.lstm(net,100,dropout = 0.7)
net = tflearn.fully_connected(net,2,activation = 'softmax')
net = tflearn.regression(net,optimizer = 'adam',learning_rate = 0.1,loss = 'categorical_crossentropy')

model = tflearn.DNN(net,tensorboard_verbose = 0)
model.fit(train_X,train_y,validation_set = (test_X,test_y),n_epoch = 20,show_metric = 32,batch_size = 32)`

Output:
![amazon_score](https://user-images.githubusercontent.com/26377913/29250893-6343c396-8068-11e7-81f1-d526914d571d.png)

Dataset:
[amazon_cells_labelled.txt](https://github.com/tflearn/tflearn/files/1220532/amazon_cells_labelled.txt)

**Comment by Noahyt** [comment]
What are actual predictions made by the network --  an accuracy of ~50% typically means the network is predicting the same class for every input.  Since (I believe) the predictions of the network are binary for this data set a 50% accuracy would correspond to predicting a constant value.

**Comment by adimyth** [comment]
Agreed, I feel the same,that the Neural Net isn't learning something.
Yeah,it's a binary classification problem. But I can't find the possible reason for this behaviour.Since,it's just my second LSTM model, I am not too sure,if everything I coded is correct. I feel it must be related to something in the preprocessing step.Can u tell me what could have possibly gone wrong? I have been banging my head over it for a week.

**Comment by AlibekJ** [comment]
My network started showing signings of learning at about 500 epohs.
More or less converged at about 3000.

Just saying.

**Comment by adimyth** [comment]
Ok,I'll surely try that!! Any other suggestions? @AlibekJ

**Comment by AlibekJ** [comment]
Would suggest creating a stupid simple dataset to verify that your setup works. Then use real data.
For example, if you are making an image classifier, feed black and white images. Something obvious and stupid simple. Then your network will converge in few epochs, then you will verify it works.

**Comment by minhajfalaki** [comment]
hey @adimyth have you got it corrected?
